File: 0791-custom-sort-string/0791-custom-sort-string.cpp
class Solution {
public:
    string customSortString(string order, string s) {
        
        unordered_map<char, int> map;
        for(int i =0; i<order.size(); i++){
            map[order[i]] = i;
        }
        
        int val = order.size();
        for(int i=0; i<s.size(); i++){
            if(map.count(s[i]) == 0){
                map[s[i]] = val++;
            }
        }
        sort(s.begin(), s.end(), [&map](char c1, char c2) {
            return map[c1] < map[c2];
        });
//         sort(s.begin(), s.end(), [](char &c1, char &c2){
//             return map[c1] < map[c2];
            
//         });
        
        return s;
        
    }
};

File: 0791-custom-sort-string/0791-custom-sort-string.py
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        
        
        dic = {}
        
        for ch in s:
            if ch in dic:
                dic[ch] += 1
            else:
                dic[ch] = 1
        
        
        ans = ""
        
        for ch in order:
            if ch in dic:
                ans += (ch*dic[ch])
                del dic[ch]
        
        if dic:
            for ch, v in dic.items():
                ans += ch*v
        
        return ans